### PR TITLE
Fixed #27217 -- Fixed CreateModel optimization with models that use mixins.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/27217

### CODE REVIEW

**Comment by charettes** [comment]
Is there a reason not to test `CreateModel.references_model` directly?

File: tests/migrations/test_optimizer.py [hunk 1]
```diff
@@ -301,6 +301,39 @@ def test_create_model_add_field(self):
...
+                    fields=[("name", models.CharField(max_length=255))],
+                    bases=(Mixin, UnicodeModel),
+                ),
+            ],
+            [
+                migrations.CreateModel(
+                    name="Foo",
+                    fields=[
+                        ("name", models.CharField(max_length=255)),
+                    ],
```
Comment (charettes) [inline tests/migrations/test_optimizer.py]: Use the same indentation as other operations for `fields`?

**Comment by timgraham** [comment]
I pushed an alternate test and will remove the other one if you prefer. I wasn't sure about the best way to test it since I haven't seen any unit tests for operation methods.

**Comment by charettes** [comment]
The patch and the new test LGTM.

**Pull request merged**